sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/credential/TokenSupplier.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.credential;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import reactor.core.publisher.Mono;
+import com.azure.core.credential.AccessToken;
+
+
+public final class TokenSupplier<T extends TokenCredential, R extends TokenRequestContext> {
```

Comment:
The need for this class has some redundancy. 
Polymorphism should take care of things when classes are passed into the normal constructors consuming TokenCredential and TokenRequestContext.